### PR TITLE
HeaderControl update style, UI Adjustments Integrations

### PR DESCRIPTION
**Pull request by TackAdam**
### Description
1. Update plugin header control to use functionality for styling from core.
2. Adjust the descriptions and links to use new header control functionality in notebooks and integrations.
3. Small adjustment to applications to move button to header for the old navigation.
4. Update integrations Cypress testing for new delete (just requires standard "delete" instead of application name following use pattern applied everywhere else).
5. Update Integrations UI

~2 New Header control change (Buttons and Description field)
| Before | After |
| ------ | ----- |
| NotebookBefore | NotebookAfter |
| IntegBefore | IntegAfter |

~4 Updated Cypress Testing
IntegCypress

~5 Integration Changes
a. Image on cards smaller
5aCards
b. Bulk delete
5bBulkDelete
c. Empty State
5cEmptyState
d. Underline Tabs
5dUnderlineTabs
e. Padding to 16px
5ePaddingAroundTable
f. Delete button secondary /hover
5fDeleteButtonSecondary:Hover
g. Panel added to Template text
5gPanelAdded
h. Remove Details / space between check new version
5hRemoveDetails:spaceOnNewVersions
i. Fix spelling
5iFixSpelling
h. Headings all adjusted to H3
5jH3HeadingsAllTitles

Integrations Before:
Screenshot 2024-09-17 at 11 47 21 AM
Screenshot 2024-09-17 at 11 43 59 AM
Screenshot 2024-09-17 at 11 44 09 AM
Screenshot 2024-09-17 at 11 44 24 AM

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/getting_started/components/getting_started_integrationCards.tsx [hunk 1]
```diff
@@ -136,13 +136,13 @@ export const IntegrationCards = () => {
               icon={
                 integration.statics && integration.statics.logo && integration.statics.logo.path ? (
                   <img
-                    style={{ height: 100, width: 100 }}
+                    style={{ height: 60, width: 60 }}
```
Comment (ps48) [inline public/components/getting_started/components/getting_started_integrationCards.tsx:139]: For the cards, I see the custom height and width. I see these are shared between getting started and integrations may be we can create a custom class in scss so both of them consume it directly. Can be added here: https://github.com/opensearch-project/dashboards-observability/blob/main/public/index.scss

File: public/components/getting_started/components/getting_started_integrationCards.tsx [hunk 2]
```diff
@@ -136,13 +136,13 @@ export const IntegrationCards = () => {
...
                     alt=""
                     className="synopsisIcon"
                     src={`/api/integrations/repository/${integration.name}/static/${integration.statics.logo.path}`}
                   />
                 ) : (
-                  <div style={{ height: 100, width: 100, backgroundColor: '#ccc' }} />
+                  <div style={{ height: 60, width: 60, backgroundColor: '#ccc' }} />
```
Comment (ps48) [inline public/components/getting_started/components/getting_started_integrationCards.tsx:145]: same as above

File: public/components/integrations/components/added_integration_table.tsx [hunk 3]
```diff
@@ -100,36 +84,38 @@ export function AddedIntegrationsTable(props: AddedIntegrationsTableProps) {
...
+    const deletePromises = selectedItems.map(async (item) => {
+      try {
+        await http.delete(`${INTEGRATIONS_BASE}/store/${item.id}`);
+        setToast(`${item.name} integration successfully deleted!`, 'success');
+      } catch (error) {
+        setToast(`Error deleting ${item.name} or its assets`, 'danger');
+      }
+    });
+
+    await Promise.all(deletePromises);
```
Comment (ps48) [inline public/components/integrations/components/added_integration_table.tsx:97]: do we need this line? I see an await on line number 90

File: public/plugin_helpers/plugin_headerControl.tsx [hunk 4]
```diff
@@ -23,48 +32,104 @@ export const HeaderControlledComponentsWrapper = ({
...
+                  };
+                } else if ((component as TopNavControlLinkData).controlType === 'link') {
+                  const linkData = component as TopNavControlLinkData;
+                  return {
+                    label: linkData.label,
+                    href: linkData.href,
+                    target: linkData.target,
+                    controlType: 'link',
+                  };
+                }
```
Comment (ps48) [inline public/plugin_helpers/plugin_headerControl.tsx:117]: we should move it to a function and add switch case to check component type

**Review by ps48 — APPROVED** [review]

File: public/components/integrations/components/added_integration_table.tsx (hunk 3 above)
Comment (TackAdam) [reply public/components/integrations/components/added_integration_table.tsx:97]: Removed

File: public/components/getting_started/components/getting_started_integrationCards.tsx (hunk 2 above)
Comment (TackAdam) [reply public/components/getting_started/components/getting_started_integrationCards.tsx:145]: Added

```
.synopsisIcon {
    height: 40px;
    width: 40px;
  }
```

File: public/components/getting_started/components/getting_started_integrationCards.tsx (hunk 1 above)
Comment (TackAdam) [reply public/components/getting_started/components/getting_started_integrationCards.tsx:139]: Added

```
.synopsisIcon {
    height: 40px;
    width: 40px;
  }
```

**Review by Swiddis — APPROVED** [review]

**Pull request merged**